storagedriver/factory/factory.go

```diff
@@ -58,7 +58,12 @@ func Create(name string, parameters map[string]interface{}) (storagedriver.Stora
 		// }
 		// return driverClient, nil
 	}
-	return driverFactory.Create(parameters)
+	d, err := driverFactory.Create(parameters)
```

Comment:
Then you can't really enforce usage of this (will be prone to errors)  and reintroduce the duplication. :-)  this should stay in the factory in my opinion.